core/silkworm/execution/processor.cpp

```diff
@@ -146,24 +148,26 @@ uint64_t ExecutionProcessor::refund_gas(const Transaction& txn, uint64_t gas_lef
 }
 
 ValidationResult ExecutionProcessor::execute_block_no_post_validation(std::vector<Receipt>& receipts) noexcept {
-    receipts.clear();
-    receipts.reserve(evm_.block().transactions.size());
+    const Block& block{evm_.block()};
 
-    uint64_t block_num{evm_.block().header.number};
-    if (block_num == evm_.config().dao_block) {
+    if (block.header.number == evm_.config().dao_block) {
         dao::transfer_balances(state_);
     }
 
     cumulative_gas_used_ = 0;
-    for (const Transaction& txn : evm_.block().transactions) {
+
+    receipts.resize(block.transactions.size());
+    auto receipt_it{receipts.begin()};
+    for (const auto& txn : block.transactions) {
         const ValidationResult err{validate_transaction(txn)};
         if (err != ValidationResult::kOk) {
             return err;
         }
-        receipts.push_back(execute_transaction(txn));
+        execute_transaction(txn, *receipt_it);
```

Comment:
Shouldn't we check, after return, for `txn.success == true` ?

Reply:
Nope: blocks with unsuccessful transactions are still valid. E.g. https://etherscan.io/tx/0xeb1272b31fb9ef4a876c3a386748d0b3fcf76d7a3aac280336fdbecffa027f99

Reply:
Ok